prereise/gather/hydrodata/eia/decompose_profile.py

```diff
@@ -80,3 +82,25 @@ def get_profile_by_plant(plant_df, total_profile):
         plant_profile = [val * factor for val in total_profile]
         res_profile[plantid] = plant_profile.copy()
     return res_profile
+
+
+def get_normalized_profile(plant_df, plant_profile):
+    """Normalize plant level profile by Pmax of the corresponding plant
+
+    :param pandas.DataFrame plant_df: plant dataframe contains generator capacity as
+        'Pmax' for each entry.
+    :param pandas.DataFrame plant_profile: plant level profile with values of generation
+    :return: (*pandas.DataFrame*) -- normalized plant level profile
+    :raises TypeError: if plant_df and/or plant_profile is not a pandas.Dataframe
+    :raises ValueError: if plant_df does not contain 'Pmax' as a column and/or
+        plants in plant_df don't match the ones in plant_profile
+    """
+    if not isinstance(plant_df, pd.DataFrame):
+        raise TypeError("plant_df must be a pandas.DataFrame object")
+    if not isinstance(plant_profile, pd.DataFrame):
+        raise TypeError("plant_profile must be a pandas.DataFrame object")
+    if "Pmax" not in plant_df.columns:
+        raise ValueError("Pmax must be one of the columns of plant_df")
+    if sorted(plant_df.index.tolist()) != sorted(plant_profile.columns.tolist()):
```

Comment:
`sorted(plant_df.index.tolist())` -> `set(plant_df.index)`? (and the same for the profile columns)

Reply:
Done.